### PR TITLE
oceanbench broadcast: remove templates from the js and separate the js from the HTML

### PR DESCRIPTION
**Pull request by trekhopton**
This was done as an initial step for getting templates out of the javascript. We shouldn't use templates in javascript because it upsets HTML and JS formatters.

We also no longer need StartTime and EndTime in the broadcast config. There are already two other fields that store this information. The javascript can just use the timestamps.

### CODE REVIEW

File: cmd/oceanbench/main.go [hunk 1]
```diff
@@ -168,6 +168,16 @@ var templateFuncs = template.FuncMap{
 	"localdate":     formatLocalDate,
 	"localtime":     formatLocalTime,
 	"localdatetime": formatLocalDateTime,
+	"json":          toJSON,
+}
+
+func toJSON(v interface{}) string {
```
Comment (scruzin) [inline cmd/oceanbench/main.go:174]: Please move this function into utils.go

File: cmd/oceantv/broadcast.go [hunk 2]
```diff
@@ -97,10 +97,8 @@ type BroadcastConfig struct {
 	Description       string        // The broadcast description shown below viewing window.
 	Privacy           string        // Privacy of the broadcast i.e. public, private or unlisted.
 	Resolution        string        // Resolution of the stream e.g. 1080p.
-	StartTime         string        // Start time of the broadcast in yy/mm/dd, hh:mm format.
 	StartTimeUnix     string        // Start time of the broadcast in unix format.
```
Comment (scruzin) [inline cmd/oceantv/broadcast.go:100]: Prefer Timestamp over TimeUnix.

**Review by scruzin — CHANGES_REQUESTED** [review]

**Comment by saxon-milton** [comment]
> This was done as an initial step for getting templates out of the javascript. We shouldn't use templates in javascript because it upsets HTML and JS formatters.
> 
> We also no longer need StartTime and EndTime in the broadcast config. There are already two other fields that store this information. The javascript can just use the timestamps.

Are there other reasons why we shouldn't use templates in HTML and JS formatters ? Because I don't think this is a very good one.

File: cmd/oceanbench/main.go (hunk 1 above)
Comment (saxon-milton) [reply cmd/oceanbench/main.go:174]: This could also be called "marshal", but up to you.

**Review by saxon-milton — APPROVED** [review]

**Review by ao-david — APPROVED** [review]

**Comment by trekhopton** [comment]
> > This was done as an initial step for getting templates out of the javascript. We shouldn't use templates in javascript because it upsets HTML and JS formatters.
> > We also no longer need StartTime and EndTime in the broadcast config. There are already two other fields that store this information. The javascript can just use the timestamps.
> 
> Are there other reasons why we shouldn't use templates in HTML and JS formatters ? Because I don't think this is a very good one.

HTML templates belong in HTML, not javascript. We could hypothetically use templates in any code but it's not good practice imo and there isn't much support for that sort of thing. Not being able to format HTML code is just a hindrance to development work flow. With a little more effort we can keep templates in the HTML and things are much neater.

File: cmd/oceantv/broadcast.go (hunk 2 above)
Comment (trekhopton) [reply cmd/oceantv/broadcast.go:100]: I'll do this in another PR.

File: cmd/oceanbench/main.go (hunk 1 above)
Comment (trekhopton) [reply cmd/oceanbench/main.go]: I moved it to utils. True, I prefer toJSON.

**Pull request merged**